pkg/arvo/gen/validate-ship.hoon

```diff
@@ -0,0 +1,108 @@
+::  Validate userspace state
+::
+::  /hoon/validate-ship/gen
+::
+/+  resource
+:-  %say
+|=  $:  [now=@da eny=@uvJ =beak]
+        [~ ~]
+    ==
+:-  %noun
+=|  issues=(list tape)
+|^
+=.  issues
+  check-group-syncs
+issues
+::
+++  scry-synced
+  |=  app=term
+  ^-  (set path)
+  %~  key  by
+  .^  (map path ship)
+    %gx
+    (scot %p p.beak)
+    app
+    (scot %da now)
+    /synced/noun
+  ==
+::
+++  scry-grp-push-syncs
+  ^-  (set resource)
+  .^  (set resource) 
+    %gx
+    (scot %p p.beak)
+    %group-push-hook
+    (scot %da now)
+    /synced/noun
+  ==
+::
+++  scry-grp-pull-syncs
+  ^-  (set resource)
+  %~  key  by
+  .^  (map resource ship)
+    %gx
+    (scot %p p.beak)
+    %group-pull-hook
+    (scot %da now)
+    /synced/noun
+  ==
+::
+++  report-missing-syncs
+  |*  [app=term syncs=(set)]
+  ^+  issues
+  %+  weld
+    issues
+  %+  turn
+    ~(tap in syncs)
+  |=(thing=* "{<app>}-hook missing: {<thing>}")
+::  +check-group-syncs: validate sync for group
+::
+::    ensure that syncs are correctly set up for each
+::    hook that has group associated resources
+++  check-group-syncs
+  ^+  issues
+  =/  groups=(set resource)
```

Comment:
I suppose this isn't the place to discuss this, but I'm very surprised group-store is still a `(map path group` instead of `(map resource group)`. I can't believe I didn't catch it in review... This sort of magic shouldn't be necessary to get a `(set resource)`.

Reply:
It's because a %gy scry must produces an `arch` i.e. `[dir=(map @ta ~) fil=(unit @uvI)]` so we encode the resource into a path and then encode it into a cord. Internally, group-store still uses a `(map resource group)`

Reply:
> a %gy scry must produces an arch

Is this actually enforced? I don't think it is. Not convinced `arch` exactly the right structure for gall `%y` scries, or at the very least it exposes more weirdness around using paths in paths. "encode into path into cord" (and then back again) gives me bad vibes.

Reply:
Apparently not, I didn't know that. Will change to a `(set resource)`